### PR TITLE
Integrate Exousia 1.0.0 M2 and adjust enterprise beans code

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

### CODE REVIEW

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java [hunk 1]
```diff
@@ -119,110 +114,84 @@
...
         STATELESS, STATEFUL, SINGLETON, MESSAGE_DRIVEN, ENTITY, READ_ONLY
     };
 
-    protected static final Logger _logger  = LogFacade.getLogger();
+    protected static final Logger _logger = LogFacade.getLogger();
 
-    @LogMessageInfo(
-        message = "The feature {0} requires Full Java EE Profile to be supported",
-        level = "WARNING")
+    @LogMessageInfo(message = "The feature {0} requires Full Java EE Profile to be supported", level = "WARNING")
```
Comment (dmatej) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java:164]: The original formatting was more readable ...

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java [hunk 2]
```diff
@@ -234,33 +203,30 @@
...
+    static final int EJBLocalHome_create = 15;
 
     // true if home method, false if component intf method.
     // Used for setting info on invocation object during authorization.
-    private static final boolean[] EJB_INTF_METHODS_INFO =
-    { true,  true,  true,  true,  true,
-      false, false, false, false, false,
-      false, false, false, false,
-      true,  true };
+    private static final boolean[] EJB_INTF_METHODS_INFO = { true, true, true, true, true, false, false, false, false, false, false, false,
```
Comment (dmatej) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java:225]: I dislike previous array of booleans as I have no idea what it is, and this is probabaly even worse. But ok, the biggest issue is that it is an array of 16 booleans.

File: appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/security/application/EJBSecurityManager.java [hunk 3]
```diff
@@ -785,21 +359,19 @@ public Principal getCallerPrincipal() {
...
              * own permission cache, which must be unregistered from factory to
-             * avoid leak.
+             * avoid leaks.
              */
             PermissionCacheFactory.removePermissionCache(uncheckedMethodPermissionCache);
             uncheckedMethodPermissionCache = null;
-            roleMapperFactory.removeAppNameForContext(this.contextId);
+            roleMapperFactory.removeAppNameForContext(contextId);
 
-        } catch (PolicyContextException pce) {
```
Comment (dmatej) [inline appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/security/application/EJBSecurityManager.java:801]: This smells, catching just ISE? What about other RTE? They should or should not be propagated? At least comment why is this catch block so special would be fine.

**Review by dmatej — CHANGES_REQUESTED** [review]
See individual comments. Also some explanation would be fine, and some info about usage and testing in the PR description ;)

File: appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/security/application/EJBSecurityManager.java (hunk 3 above)
Comment (arjantijms) [reply appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/security/application/EJBSecurityManager.java:801]: The previous code caught PolicyContextException as a special case. This has been replaced by an IllegalStateException. It just maintains that behaviour.

Comment (dmatej) [reply appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/security/application/EJBSecurityManager.java:801]: Hmm, I don't like it much, but ok.

**Review by dmatej — APPROVED** [review]
I have yet another thought about @LogMessageInfo annotation. This formatting is ugly, but the level is used just for documentation purpose, which is not regenerated now (or am I wrong?) and the original work around logging never finished.
So I accept it as it is (build passed).

**Comment by arjantijms** [comment]
> I have yet another thought about @LogMessageInfo annotation. This formatting is ugly, but the level is used just for documentation purpose, which is not regenerated now (or am I wrong?) and the original work around logging never finished.

Yes, I just peeked at it. Don't know what to do with it atm.

**Pull request merged**